src/DataWrangling/JRA55.jl

```diff
@@ -159,64 +204,266 @@ function jra55_field_time_series(variable_name;
+        # interpolate to, and the following is a sketch of the JRA55 native grid,
+        #
+        #      1         2         3         4         5
+        # |         |         |         |         |         |
+        # |    x  ᵒ |    x    |    x    |    x  ᵒ |    x    |
+        # |         |         |         |         |         |
+        # 1         2         3         4         5         6 
+        #
+        # then for example, we should find that (iᵢ, i₂) = (1, 5).
+        # So we want to reduce the first index by one, and limit them
+        # both by the available data. There could be some mismatch due
+        # to the use of different coordinate systems (ie whether λ ∈ (0, 360)
+        # which we may also need to handle separately.
+        
+        i₁ = searchsortedfirst(λc, λ₁)
+        j₁ = searchsortedfirst(φc, φ₁)
+
+        i₂ = searchsortedfirst(λc, λ₂)
+        j₂ = searchsortedfirst(φc, φ₂)
+
+        i₁ = max(1, i₁ - 1)
+        j₁ = max(1, j₁ - 1)
+
+        i₂ = min(Nx, i₂)
+        j₂ = min(Ny, j₂)
+
+        TX = λ₂ - λ₁ ≈ 360 ? Periodic : Bounded
+    end
 
-    grid = LatitudeLongitudeGrid(architecture,
-                                 size = (Nx, Ny);
-                                 longitude = λ,
-                                 latitude = φ,
-                                 topology = (Periodic, Bounded, Flat))
+    # Extract variable data
+    data = ds[shortname][i₁:i₂, j₁:j₂, time_indices]
+    λr = λn[i₁:i₂+1]
+    φr = φn[j₁:j₂+1]
+    Nrx, Nry, Nt = size(data)
+
+    close(ds)
 
     # Hack together the `times` for the JRA55 dataset we are currently using.
     # We might want to use the acutal dates instead though.
-    # So the following code maybe should change.
+    # So the following code might need to change.
     Δt = 3hours # just what it is
     Nt = length(times)
     start_time = 0 # Note: the forcing start at Jan 1 of the repeat year.
     stop_time = Δt * (Nt - 1)
     times = start_time:Δt:stop_time
 
-    boundary_conditions = FieldBoundaryConditions(grid, (Center, Center, Nothing))
-    fts = FieldTimeSeries{Center, Center, Nothing}(grid, times; boundary_conditions)
+    jra55_native_grid = LatitudeLongitudeGrid(architecture,
+                                              size = (Nrx, Nry);
+                                              longitude = λr,
+                                              latitude = φr,
+                                              topology = (TX, Bounded, Flat))
 
-    # Fill the data
-    interior(fts, :, :, 1, :) .= data[:, :, :]
+    boundary_conditions = FieldBoundaryConditions(jra55_native_grid, (Center, Center, Nothing))
+
+    grid = isnothing(grid) ? jra55_native_grid : grid
+    loc  = (LX, LY, Nothing)
+
+    fts = retrieve_and_maybe_write_jra55_data(time_chunks_in_memory, grid, times, loc, boundary_conditions, data, jra55_native_grid; 
+                                              interpolated_file, shortname)
```

Comment:
Let's design a user interface that generalizes how the user specifies `time_indices` (one of the arguments to `jra55_field_time_series` so that the user can say "load these indices to disk, leave the rest in a file" or something like that.